File: backend/app/schemas/analytics.py
from decimal import Decimal

from pydantic import BaseModel


class MonthlyAnalytics(BaseModel):
    month: str
    income: Decimal
    expenses: Decimal


class CategoryAnalytics(BaseModel):
    category_name: str
    amount: Decimal


class AnalyticsSummary(BaseModel):
    monthly: list[MonthlyAnalytics]
    categories: list[CategoryAnalytics]

File: frontend/src/api/Analytics.ts
import apiClient from "./client";

export interface MonthlyAnalytics {
  month: string;
  income: string;
  expenses: string;
}

export interface CategoryAnalytics {
  category_name: string;
  amount: string;
}

export interface AnalyticsSummary {
  monthly: MonthlyAnalytics[];
  categories: CategoryAnalytics[];
}

export async function getAnalyticsSummary(): Promise<AnalyticsSummary> {
  const response = await apiClient.get<AnalyticsSummary>(
    "/analytics/summary",
  );

  return response.data;
}